Backpack-SwiftUI/TextField/Classes/BPKTextField.swift

```diff
@@ -0,0 +1,110 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2022 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+/// A control that displays an editable text interface.
+/// 
+/// Use `inputState(_ state: State)` to change the state of the text field.
+public struct BPKTextField: View {
+    struct Icon {
+        let icon: BPKIcon
+        let color: BPKColor
+    }
+    
+    @Binding private var text: String
+    private let placeholder: String
+    private var state: State = .default
+
+    /// Creates a `BPKTextField`.
+    ///
+    /// - Parameters:
+    ///   - placeholder: The placeholder text to display when the text field is empty.
+    ///   - text: The text to display in the text field.
+    public init(
+        placeholder: String = "",
+        _ text: Binding<String>
+    ) {
+        self.placeholder = placeholder
+        self._text = text
+    }
+    
+    public var body: some View {
+        TextField(placeholder, text: $text)
+            .font(style: .bodyDefault)
+            .foregroundColor(state.textColor)
+            .padding(.md)
+            .background(.surfaceDefaultColor)
+            .clipShape(RoundedRectangle(cornerRadius: .sm))
+            .outline(state.borderColor, cornerRadius: .sm)
+            .overlay(accessory)
+            .disabled(state.isDisabled)
+    }
+    
+    private var accessory: some View {
+        HStack {
+            Spacer()
+            if let icon = state.icon {
+                if case let .clear(accessibilityLabel, action) = state {
+                    Button(action: action) {
+                        BPKIconView(icon.icon)
+                            .foregroundColor(icon.color)
+                    }
+                        .accessibilityLabel(accessibilityLabel)
```

Comment:
Focus order on this button is still wrong (Clear first, then field). It should focus on the field and then the clear button.

Sent you something on Slack on how we could achieve this in a different way, with correct order.